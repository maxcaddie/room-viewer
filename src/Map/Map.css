.navigationArrows {
  color: red;
  height: 70vh;
  width: 10vw;
  font-size: 20vh;
  background-color: transparent;
}

.floorMap {
  width: 70vw;
  min-width: 600px;
  height: 80vh;
}
