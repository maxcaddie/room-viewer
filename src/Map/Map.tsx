import React from "react"

const Map: React.FC = () => {
    return (
      <React.Fragment>
        <p className="floor">This will say what floor you are looking at</p>
        <img />
      </React.Fragment>);
  }
  
  export default Map;