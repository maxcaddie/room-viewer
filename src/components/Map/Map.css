.container {
  width: auto;
  min-width: 600px;
  height: 90vh;
  position: relative;
  margin-top: 75px;
}

.navigationArrows {
  height: 100%;
  width: 50%;
  background-color: transparent;
  border-color: transparent;
  bottom: 0px;
  outline: none !important;
}

.left {
  position: absolute;
  left: 0px;
}

.right {
  position: absolute;
  right: 0px;
}

.floorMap {
  width: 100%;
  max-width: 110vh;
  height: 95%;
  max-height: 110vw;
}

.floor {
  font-family: "Adelle, Arial";
  color: white;
  font-size: 20px;
}
