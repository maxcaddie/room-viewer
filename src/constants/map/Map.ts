export const LEFT_ARROW_KEY_CODE = 37;
export const RIGHT_ARROW_KEY_CODE = 39;
export const A_KEY_CODE = 37;
export const D_KEY_CODE = 39;

export function isGoBack(keyCode: number) {
  return keyCode === LEFT_ARROW_KEY_CODE || keyCode === A_KEY_CODE;
}

export function isGoForward(keyCode: number) {
  return keyCode === RIGHT_ARROW_KEY_CODE || keyCode === D_KEY_CODE;
}
