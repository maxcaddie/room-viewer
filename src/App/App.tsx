//Internals
import Map from "../Map/Map";

//Externals
import React from "react";

//Stylesheets
import "./App.css";

const App: React.FC = () => {
  return (
    <div className="App">
      <Map />
    </div>
  );
};

export default App;
