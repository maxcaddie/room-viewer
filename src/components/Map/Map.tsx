//Internals
import { store } from "../../State";
import { nextFloor, prevFloor } from "../../redux/actions/floorActions";
import { numToFloor, numToFloorName } from "../../constants/floor/Floor";
import { isGoBack, isGoForward } from "../../constants/map/Map";

//Externals
import React, { useEffect } from "react";
import { useSelector } from "react-redux";

//Stylesheets
import "./Map.css";
import { MOBILE_SIZE } from "../../constants/MobileView";

const Map: React.FC = () => {
  useEffect(() => {
    document.onkeydown = function(e) {
      e = e || window.event;
      if (isGoBack(e.keyCode)) {
        store.dispatch(prevFloor());
      }
      if (isGoForward(e.keyCode)) {
        store.dispatch(nextFloor());
      }
    };
  });

  const floorOn = useSelector(store.getState).floor;

  return (
    <React.Fragment>
      <style>{"body { background-color: #e9ebef; }"}</style>
      {/* <p className="floor">{numToFloorName(floorOn)}</p> */}
      <div className="container">
        <img src={numToFloor(floorOn)} alt={numToFloor(floorOn)} className="floorMap" />
        <button
          className="navigationArrows left"
          onClick={() => {
            window.innerWidth > MOBILE_SIZE ? store.dispatch(prevFloor()) : store.dispatch(nextFloor());
          }}
        />
        <button className="navigationArrows right" onClick={() => store.dispatch(nextFloor())} />
      </div>
    </React.Fragment>
  );
};

export default Map;
