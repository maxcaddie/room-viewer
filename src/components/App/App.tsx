//Internals
import Map from "../Map/Map";
import Header from "../Header/Header";

//Externals
import React from "react";

//Stylesheets
import "./App.css";

const App: React.FC = () => {
  return (
    <div className="App">
      <Header />
      <Map />
    </div>
  );
};

export default App;
