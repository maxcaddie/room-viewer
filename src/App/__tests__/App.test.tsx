//Internals
import App from '../App';
import Map from '../../Map/Map'

//Externals
import React from 'react';
import renderer from 'react-test-renderer';



it('renders app correctly',()=>{
    const appComponent = renderer.create(<App />).toJSON();
    expect(appComponent).toMatchSnapshot();
});

it('the outermost div has the className app',()=>{
    const appComponent = renderer.create(<App />);
    const element = appComponent.root.findByType("div");
    expect(element.props.className.includes("App")).toBeTruthy();
});

it('includes a map component',()=>{
    const appComponent = renderer.create(<App />);
    appComponent.root.findByType(Map);
})