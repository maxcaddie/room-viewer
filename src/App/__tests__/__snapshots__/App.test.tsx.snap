// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app correctly 1`] = `
<div
  className="App"
>
  <p
    className="floor"
  >
    This will say what floor you are looking at
  </p>
  <img />
</div>
`;
